ir_aarch64.dasc

```diff
@@ -4955,13 +4975,8 @@ static void ir_emit_tailcall(ir_ctx *ctx, ir_ref def, ir_insn *insn)
 			|	br Rx(IR_REG_INT_TMP)
 		}
     } else {
-		ir_reg op2_reg = ctx->regs[def][2];
-
 		IR_ASSERT(op2_reg != IR_REG_NONE);
-		if (IR_REG_SPILLED(op2_reg)) {
-			op2_reg = IR_REG_NUM(op2_reg);
-			ir_emit_load(ctx, IR_ADDR, op2_reg, insn->op2);
-		}
```

Comment:
Can we have a spilled ``op2_reg`` that not interference with  ``ctx->used_preserved_reg``? 
I think you lost this case.

Reply:
Yes, I missed this case. I've updated the PR and added tests for that.